lib/api/dsl/methods.js

```diff
@@ -690,15 +760,17 @@ module.exports = methods = {
     }
 
     try {
-      /*eslint guard-for-in: 0 */
       for (i in Object.keys(filter)) {
```

Comment:
array.foreach is asynchronious, and we need to have something synchronious here.

Two cases: 
* transform the ```if (!Object.keys(filter).hasOwnProperty(i)) {
   continue;
}``` into ``` if (Object.keys(filter).hasOwnProperty(i)) { ... ```
* keep the guard-for-in exception since it is an array, and for in statements will only get array members...